app.rb

```diff
@@ -157,6 +164,55 @@ def check_date(date)
     false
   end
 
+  def music_display
+    list_music
+  end
+
+  def genre_display
+    list_genre
+  end
+
+  def music_create
+    create_music
+  end
+
+  def save_files
+    instance_variables.each do |var|
+      file_name = var.to_s.chomp('_list').delete('@')
+      ary = []
+      instance_variable_get(var).each do |obj|
+        hash = { ref: obj, value: to_hash(obj) }
+        ary << hash
+      end
+      File.write("./data/#{file_name}.json", JSON.generate(ary))
+    end
+  end
+
+  def read_files
+    instance_variables.each do |var|
+      file_name = var.to_s.chomp('_list').delete('@')
+      if File.exist?("./data/#{file_name}.json") && File.read("./data/#{file_name}.json") != ''
+        ary = JSON.parse(File.read("./data/#{file_name}.json"))
+        case file_name
+        when 'music'
+          read_music(ary)
+        when 'genres'
+          read_genre(ary)
+        end
+      else
+        File.write("./data/#{file_name}.json", '[]')
+      end
+    end
+  end
+
```

Comment:
[OPTIONAL] Nice job looking for each file on the project and storing the information on it. 

But when you are working on a group project, maybe is better to focus on your task, it's can avoid conflicts if your partners are working on the same section but on a different task.

***For example here you are storing the information of the whole app, and your partners have to do the same thing with their files.***

So they can implement a method to store the information in a different way and if they are not noticed it can be a conflict when the data is going to be stored in the final PR.

**Good Job and thanks for worrying about the different parts of the project @masangana !**

**KEEP CODING!**